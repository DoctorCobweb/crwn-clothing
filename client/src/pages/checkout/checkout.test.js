import React from 'react';
import { shallow } from 'enzyme';

import { CheckoutPage } from './checkout.component';
import CheckoutItem from '../../components/checkout-item/checkout-item.component';

describe('Checkout component', () => {
  let wrapper;
  let mockCartItems;
  let mockTotal;

  beforeEach(() => {
    mockCartItems= [{id:0}, {id:1}, {id:2}, {id:3}, {id:4},];
    mockTotal = 10;
    const mockProps = {
      cartItems: mockCartItems,
      total: mockTotal,
    };
    wrapper = shallow(<CheckoutPage {...mockProps} />);
  });

  it('should render Checkout', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render correct number of CheckoutItem components', () => {
    expect(wrapper.find(CheckoutItem).length).toBe(mockCartItems.length);
  });

  it('should correctly set the total', () => {
    const totalText = wrapper.find('TotalContainer').childAt(0).text();
    expect(totalText).toBe(`$${mockTotal}`);
  });


});