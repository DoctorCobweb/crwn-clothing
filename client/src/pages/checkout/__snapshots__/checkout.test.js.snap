// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout component should render Checkout 1`] = `
"<CheckoutPageContainer>
  <CheckoutHeaderContainer>
    <HeaderBlockContainer>
      <span>
        Product
      </span>
    </HeaderBlockContainer>
    <HeaderBlockContainer>
      <span>
        Description
      </span>
    </HeaderBlockContainer>
    <HeaderBlockContainer>
      <span>
        Quantity
      </span>
    </HeaderBlockContainer>
    <HeaderBlockContainer>
      <span />
    </HeaderBlockContainer>
    <HeaderBlockContainer>
      <span>
        Remove
      </span>
    </HeaderBlockContainer>
  </CheckoutHeaderContainer>
  <Connect(CheckoutItem) cartItem={{...}} />
  <Connect(CheckoutItem) cartItem={{...}} />
  <Connect(CheckoutItem) cartItem={{...}} />
  <Connect(CheckoutItem) cartItem={{...}} />
  <Connect(CheckoutItem) cartItem={{...}} />
  <TotalContainer>
    <span>
      $
      10
    </span>
  </TotalContainer>
  <WarningContainer>
    *Please the following test credit card for payments*
    <br />
    4242 4242 4242 4242 - Exp: 01/20 - CVV: 123
  </WarningContainer>
  <StripeCheckoutButton price={10} />
</CheckoutPageContainer>"
`;
